### ISSUE TITLE
Support non-pure Grid.Column valueProviders

### ISSUE THREAD

**Issue by mvysny** [mvysny/karibu-testing#124]
Usually the value providers passed into `grid.addColumn(vp)` are [pure functions](https://en.wikipedia.org/wiki/Pure_function), and such use-case is correctly supported by Karibu-Testing.

However, it's possible to write the following code which works with Vaadin but fails with Karibu:
```java
var grid = new Grid<Person>();
grid.addColumn(person -> { person.setCalculated("Foo"); return person.getCalculated(); });
grid.addSelectionChangeListener(e -> assertEquals("Foo", e.getItem().getCalculated());
grid.setItems(new Person());
_select(grid, _get(grid, 0));
```
The Karibu code will fail in `_select(grid, _get(0));` - the selection listener's assertEquals() will fail. The problem is that `_get(grid, 0)` will not run the column value providers, causing the `person.calculated` property to not to be filled in.

**Comment by mvysny** [comment]
One of the solutions would be for `_get()` to take advantage of the DataCommunicator's cache (it caches all visible objects and generates a key for them; the key is then used by client-side when a row is clicked etc). This could however break existing tests which rely on getting fresh values from the data provider? However, it could be possible for `DataProvider._get()` to return fresh stuff while `Grid._get()` could return cached stuff, passed through value providers?

Another solution would be to add a function to Karibu which would run value providers on a bean, therefore shifting the responsibility to the test writer... Not sure about this.

**Comment by mvysny** [comment]
Yup, the KeyMapper is not populated. Perhaps `_get()` could populate the KeyMapper manually, then reuse the returned beans?

**Comment by mvysny** [comment]
The KeyMapper caching solution has been implemented in Karibu-Testing 1.3.18. It's a potentially breaking change, but everything happens under the hood so hopefully it won't break anything. If it does, please open a new ticket.
